### PR TITLE
Discover all routes of a service 

### PR DESCRIPTION
**Pull request by dgghinea**

### CODE REVIEW

**Comment by alrosca** [comment]
Update readme too, the local path download, part to reflect where the tags are configured

File: pkg/gateway/client.go [hunk 1]
```diff
@@ -106,64 +106,62 @@ func (gc *Client) processKongServicesList(ctx context.Context, services []*klib.
...
+	log := gc.logger.WithField("route-id", *route.ID)
+	proxyHost := gc.kongGatewayCfg.Proxy.Host
+	httpPort := gc.kongGatewayCfg.Proxy.Port.HTTP
+	httpsPort := gc.kongGatewayCfg.Proxy.Port.HTTPS
 
 	apiPlugins, err := gc.plugins.GetEffectivePlugins(*route.ID, *service.ID)
 	if err != nil {
-		log.WithError(err).Errorf("failed to get plugins for route")
-		return err
+		log.Infof("Could not list plugins for serviceID: %s, with routeID: %s", *service.ID, *route.ID)
```
Comment (jcollins-axway) [inline pkg/gateway/client.go:131]: Can you take all the values in your formatted string and place them in fields? The route ID is already a field. 

Also, should this be a Warn at least?

File: pkg/gateway/client.go [hunk 2]
```diff
@@ -106,64 +106,62 @@ func (gc *Client) processKongServicesList(ctx context.Context, services []*klib.
...
+	for _, route := range routes {
+		gc.specPreparation(ctx, route, service)
 	}
+	return nil
+}
 
-	route := routes[0]
-	log = log.WithField("route-id", *route.ID)
+func (gc *Client) specPreparation(ctx context.Context, route *klib.Route, service *klib.Service) {
+	log := gc.logger.WithField("route-id", *route.ID)
```
Comment (jcollins-axway) [inline pkg/gateway/client.go:124]: Most places we use camelCase for the field name, could you update this to routeID?

File: pkg/gateway/client.go [hunk 3]
```diff
@@ -106,64 +106,62 @@ func (gc *Client) processKongServicesList(ctx context.Context, services []*klib.
 
 func (gc *Client) processSingleKongService(ctx context.Context, service *klib.Service) error {
 	log := gc.logger.WithField("service-name", *service.Name)
```
Comment (jcollins-axway) [inline pkg/gateway/client.go:108]: Looks like I missed this before, would you mind replacing all uses of fields to camelCase?

File: pkg/gateway/client.go [hunk 4]
```diff
@@ -106,64 +106,62 @@ func (gc *Client) processKongServicesList(ctx context.Context, services []*klib.
...
-
 	oasSpec := Openapi{
 		spec: string(kongServiceSpec),
 	}
 
 	endpoints := gc.processKongRoute(proxyHost, oasSpec.BasePath(), route, httpPort, httpsPort)
 	serviceBody, err := gc.processKongAPI(ctx, *route.ID, service, oasSpec, endpoints, apiPlugins)
 	if err != nil {
-		return err
+		log.WithError(err).Error("Failed to process kong API")
```
Comment (jcollins-axway) [inline pkg/gateway/client.go:151]: error logs should start lower case, "failed..."

File: pkg/kong/kongclient.go [hunk 5]
```diff
@@ -129,21 +130,20 @@ func (k KongClient) GetSpecForService(ctx context.Context, service *klib.Service
 	return k.getSpecFromBackend(ctx, backendURL)
 }
 
-func (k KongClient) getSpecFromLocal(ctx context.Context, service *klib.Service) ([]byte, error) {
-	log := k.logger.WithField("serviceID", service.ID).WithField("serviceName", service.Name)
-	log.Info("getting spec from local storage")
+func (k KongClient) getSpecFromLocal(ctx context.Context, service *klib.Service, route *klib.Route) ([]byte, error) {
+	log := k.logger.WithField("serviceName", *service.Name).WithField("routeName", *route.Name)
```
Comment (jcollins-axway) [inline pkg/kong/kongclient.go:134]: The constant uses of the same fields, these field names should probably be made into constants

**Pull request closed without merging**